### PR TITLE
feat: add secq256k1

### PR DESCRIPTION
**Pull request by dmpierre**
This PR adds secq256k1. It is a direct inspiration from how grumpkin was implemented using bn256.

1. Adds a secq256k1 folder containing `curve.rs` and `mod.rs`
2. Updates `lib.rs` accordingly

### CODE REVIEW

File: src/secq256k1/curve.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    fn is_torsion_free(&self) -> Choice {
+        1.into()
+    }
+}
+
+impl Secq256k1 {
+    const SVDW_Z: Fq = Fq::ONE;
+}
+
+#[cfg(test)]
```
Comment (CPerezz) [inline src/secq256k1/curve.rs:83]: Seeing that with the wrong params the tests keep passing, maybe we should add some tests that check basic arithmetic curve ops to be sure all is correct.

**Review by CPerezz — CHANGES_REQUESTED** [review]

File: src/secq256k1/mod.rs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+mod curve;
+
+pub use crate::secp256k1::{Fp as Fq, Fq as Fp};
```
[the remark this answers is not in the document]
Comment (dmpierre) [reply src/secq256k1/mod.rs:3]: Yes! it's just very practical to simply invert those for `secq`. I've taken inspiration from `src/grumpkin/mod.rs`

File: src/secq256k1/curve.rs (hunk 1 above)
Comment (dmpierre) [reply src/secq256k1/curve.rs:83]: happy to help implement those if you have any ideas in mind!

**Comment by dmpierre** [comment]
@CPerezz also added hash to curve for both [secq](https://github.com/dmpierre/halo2curves/blob/deb357b0f7b510da3720f36b887003ea4b27b400/src/secq256k1/curve.rs#L52) and [secp](https://github.com/dmpierre/halo2curves/blob/deb357b0f7b510da3720f36b887003ea4b27b400/src/secp256k1/curve.rs#L68). 
From my understanding, this is using the Shallue-van de Woestijne (SW) method, which works with any ec - see [here](https://geometry.xyz/notebook/Hashing-to-the-secp256k1-Elliptic-Curve).
I've taken the `SVDW_Z` constant from [grumpkin](https://github.com/dmpierre/halo2curves/blob/deb357b0f7b510da3720f36b887003ea4b27b400/src/grumpkin/curve.rs#L80) and specified it for both [secq](https://github.com/dmpierre/halo2curves/blob/deb357b0f7b510da3720f36b887003ea4b27b400/src/secq256k1/curve.rs#L80) and [secp](https://github.com/dmpierre/halo2curves/blob/deb357b0f7b510da3720f36b887003ea4b27b400/src/secp256k1/curve.rs#L72). Since it's tentative, wouldn't mind your opinion on this :)

File: src/secq256k1/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (CPerezz) [reply src/secq256k1/mod.rs:3]: Fair I see. ! I'm fine with it, I've checked a bit more and indeed makes sense!

File: src/secq256k1/curve.rs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    |curve_id, domain_prefix| svdw_hash_to_curve(curve_id, domain_prefix, Secq256k1::SVDW_Z),
+);
+
+impl CurveAffineExt for Secq256k1Affine {
+    batch_add!();
+
+    fn into_coordinates(self) -> (Self::Base, Self::Base) {
+        (self.x, self.y)
+    }
+}
```
Comment (CPerezz) [inline src/secq256k1/curve.rs:61]: This has just been removed. So we should rebase and remove the implementation of `CurveAffineExt`

**Review by han0110 — COMMENTED** [review]
LGTM, just need a rebase.

The chosen `SVDW_Z` (both `1`) is checked locally with the script https://www.ietf.org/id/draft-irtf-cfrg-hash-to-curve-16.html#appendix-H.1

File: src/secq256k1/curve.rs (hunk 3 above)
Comment (dmpierre) [reply src/secq256k1/curve.rs]: yup, [done](https://github.com/privacy-scaling-explorations/halo2curves/pull/65/commits/b0e548f39e25cc4e8ca27c177ecde621e873e1a2)!

**Review by CPerezz — APPROVED** [review]
LGTM

**Review by han0110 — APPROVED** [review]
LGTM

**Pull request merged**